File: src/main/scala/org/tindalos/principle/domain/analyzers/Analyzer.scala
package org.tindalos.principle.domain.analyzers

import org.tindalos.principle.domain.constraints.Constraints
import org.tindalos.principle.domain.{AnalysisInput, AnalysisResult}

trait Analyzer {

  def analyze(checkInput: AnalysisInput): AnalysisResult
  def isEnabled(constraints: Constraints): Boolean
  
  override def toString() = this.getClass().getSimpleName()

}

File: src/main/java/org/tindalos/principle/domain/analyzers/Analyzer.java
package org.tindalos.principle.domain.analyzers;

import org.tindalos.principle.domain.plan.AnalysisInput;
import org.tindalos.principle.domain.core.AnalysisResult;
import org.tindalos.principle.domain.constraints.Constraints;

/**
 * Contract for all architectural analyzers.
 */
public interface Analyzer {

    AnalysisResult analyze(AnalysisInput checkInput);

    boolean isEnabled(Constraints constraints);
}